config/src/config/logger_config.rs

```diff
@@ -10,13 +12,41 @@ pub struct LoggerConfig {
     pub is_async: bool,
     // chan_size of slog async drain for node logging.
     pub chan_size: usize,
+    // The default logging level for slog.
+    #[serde(serialize_with = "serialize_filter_level")]
+    #[serde(deserialize_with = "deserialize_filter_level")]
+    pub filter_level: FilterLevel,
 }
 
 impl Default for LoggerConfig {
     fn default() -> LoggerConfig {
         LoggerConfig {
             is_async: true,
             chan_size: 256,
+            filter_level: FilterLevel::Info,
         }
     }
 }
+
+fn serialize_filter_level<S>(filter_level: &FilterLevel, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let filter_level_string = filter_level.as_str().to_string();
+    filter_level_string.serialize(serializer)
+}
+
+fn deserialize_filter_level<'de, D>(deserializer: D) -> Result<FilterLevel, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let filter_level_string: String = Deserialize::deserialize(deserializer)?;
+    FilterLevel::from_str(&filter_level_string).map_err(|_| {
```

Comment:
1. At the moment, the error message produced looks like this (assuming I set an invalid filter level of 'INVALID LEVEL' in the config):

---- config::test::verify_default_config stdout ----
thread 'config::test::verify_default_config' panicked at 'Unable to load config: unknown variant `INVALID LEVEL`, expected one of `OFF`, `CRITICAL`, `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE` for key `logger.filter_level` at line 82 column 1', src/libcore/result.rs:1165:5

I think this is clear as it tells the user what filter levels are available and what they actually tried to use.

2. Unfortunately, I can't inline filter_level_string into the from_str() call because we need to display the filter_level_string on an error. If an error is returned by from_str(), the error doesn't actually contain any useful information (see here [1]), so we can't use map_err() to actually produce any useful information, so we need to keep it in the outer scope. I'll leave this as it is for now. 

Thanks, David! 😄 

[1] https://docs.rs/slog/2.0.12/src/slog/lib.rs.html#2006-2019